crates/red_knot_python_semantic/src/types.rs

```diff
@@ -792,13 +792,13 @@ impl<'db> Type<'db> {
                 number.to_string().into_boxed_str()
             })),
             Type::BooleanLiteral(true) => {
-                Type::StringLiteral(StringLiteralType::new(db, "True".into()))
+                Type::StringLiteral(StringLiteralType::new(db, Box::<str>::from("True")))
```

Comment:
I'm curious to know why this and other related changes are being made. I do prefer using `::from` instead of `.into` because it makes the conversion visible without peeking through the definition to know the final type.

Reply:
https://github.com/salsa-rs/salsa/pull/589 added a way to lookup interned types with borrowed values (rather than owned values). This made the `new` constructor  generic. I think I'm going to wait for https://github.com/salsa-rs/salsa/pull/594 because we can then change  `StringLiteralType::new` to be called with `"True"` and avoid allocating if the value has already been interned.